### ISSUE TITLE
Add JdbcSourceConnectConfig for incrementing.table.name and timestamp.table.name

### ISSUE THREAD

**Issue by msschroe3** [confluentinc/kafka-connect-jdbc#439]
When running the JDBC connector with a custom query that joins tables the incrementing and/or timestamp column names need to be prefixed with the table name. However, because of how the provided column names automatically get surrounded with back-tick quotes it becomes an invalid WHERE clause. Also, [an alias cannot be used in MySQL WHERE clauses](https://stackoverflow.com/a/942592/2475679).

### Example
If I setup a JDBC connector with the following configuration that has a query that joins two tables

```JSON
{
  "name": "package_events",
  "config": {
          "connector.class": "io.confluent.connect.jdbc.JdbcSourceConnector",
          
          "connection.url": "jdbc:mysql://localhost:3306/events",
          "connection.user": "root",
          "connection.password": "root",

          "query": "select e.`id`, e.`status`, e.`user_id`, e.`event_date`, p.`address`, from `event` as e join `package` as p on e.`package_id` = p.`id`",

          "mode": "incrementing",
          "incrementing.column.name": "e.`id`",

          "topic.prefix": "package-events",
  }
}
```

the prepared statement that gets created looks like

```SQL
select e.`id`, e.`status`, e.`user_id`, e.`event_date`, p.`id`
from `event` as e join `package` as p on e.`package_id` = p.`id`
where `e.`id`` > ? ORDER BY `e.`id`` ASC
```

which is invalid because of the extra set of quotes wrapping the `e.'id'` in the `where` clause.

### Proposed Solution
When a `ColumnId` is created during the connector configuration it can be passed a `tableId` that is prepended to the column name. See the snippet below taken from [`io.confluent.connect.jdbc.util.ColumnId`](https://github.com/confluentinc/kafka-connect-jdbc/blob/dbe749080fbe815e2d422538f159cc438b5c1332/src/main/java/io/confluent/connect/jdbc/util/ColumnId.java#L69)

```JAVA
public class ColumnId implements Expressable {
  // constructors and other methods redacted 

  @Override
  public void appendTo(ExpressionBuilder builder, boolean useQuotes) {
    if (tableId != null) {
      builder.append(tableId);
      builder.appendIdentifierDelimiter();
    }
    builder.appendIdentifier(this.name, useQuotes);
  }
```

Currently, `tableId` can only be used when running in `TABLE` mode. See the constructor in [`io.confluent.connect.jdbc.source.TableQuerier`](https://github.com/confluentinc/kafka-connect-jdbc/blob/master/src/main/java/io/confluent/connect/jdbc/source/TableQuerier.java#L53)

```JAVA
  public TableQuerier(DatabaseDialect dialect, QueryMode mode, String nameOrQuery, String topicPrefix) {
    this.dialect = dialect;
    this.mode = mode;
    this.tableId = mode.equals(QueryMode.TABLE) ? dialect.parseTableIdentifier(nameOrQuery) : null;
    this.query = mode.equals(QueryMode.QUERY) ? nameOrQuery : null;
    this.topicPrefix = topicPrefix;
    this.lastUpdate = 0;
  }
```

If there was a configuration for the tableName that could be used when generating the `where` clause we would end up with properly formatted SQL. See the proposed config addition below (it would look very similar when running in timestamp mode - `timestamp.table.name`).

```JSON
{
  "mode": "incrementing",
  "incrementing.column.name": "id",
  "incrementing.table.name": "e"
}
```

This would result in the following `where` clause
```SQL
where e.`id` > ? ORDER BY e.`id` ASC
```

### Alternate Solution
Issue #270 discusses this problem as well and proposes disabling quotes altogether and has an open [PR](https://github.com/confluentinc/kafka-connect-jdbc/pull/269/files) with proposed changes.

### Workaround

The current workaround that I see for this Issue is to generate a MySQL `VIEW`, enable `table.types: VIEW` in the connector config, and run the out of the box table connector against the `VIEW` instead of using a custom query that joins multiple tables.

**Comment by marioalvial** [comment]
I'm facing the exact same problem but I'm not specifying the table prefix. I'm using incrementing mode and my query joins two tables. Both tables have the column field 'id'. So when I run my connectors I get "Ambiguous column 'id' in where clause".

**Comment by rhauch** [comment]
It is possible to bypass these quotes with #572, which was recently made and has not yet been released, though it will be in the 5.0.2, 5.1.1, and 5.2.0 releases whenever they occur. In the meantime, you can build the connector locally and try the new feature to disable quoting.

I'll mark this as closed, but feel free to reopen if you try the feature and it does not work for you.
